src/main/java/address/sync/CloudService.java

```diff
@@ -0,0 +1,394 @@
+        RawCloudResponse cloudResponse = cloud.deleteTag(addressBookName, tagName);
+        HashMap<String, Long> headerHashMap = getHashMapFromHeader(cloudResponse.getHeaders());
+        if (!isValid(cloudResponse)) {
+            return getResponseWithNoData(cloudResponse, headerHashMap);
+        }
+        return new ExtractedCloudResponse<>(cloudResponse.getResponseCode(), getRateLimitFromHeader(headerHashMap),
+                                            getRateRemainingFromHeader(headerHashMap),
+                                            getRateResetFromHeader(headerHashMap), null);
+    }
+
+    /**
+     * Gets the list of persons for addressBookName, which have been modified after a certain time
+     *
+     * Consumes 1 + floor(result size/RESOURCES_PER_PAGE) API usage
+     * @param addressBookName
+     * @return wrapped response with list of persons
+     * @throws IOException if content cannot be interpreted
+     */
+    @Override
+    public ExtractedCloudResponse<List<Person>> getUpdatedPersonsSince(String addressBookName, LocalDateTime time)
+            throws IOException {
+        RawCloudResponse cloudResponse = cloud.getUpdatedPersons(addressBookName, time.toString(), RESOURCES_PER_PAGE);
+        HashMap<String, Long> headerHashMap = getHashMapFromHeader(cloudResponse.getHeaders());
+        if (!isValid(cloudResponse)) {
+            return getResponseWithNoData(cloudResponse, headerHashMap);
+        }
+        List<CloudPerson> cloudPersons = getDataListFromBody(cloudResponse.getBody(), CloudPerson.class);
+        return new ExtractedCloudResponse<>(cloudResponse.getResponseCode(), getRateLimitFromHeader(headerHashMap),
+                                            getRateRemainingFromHeader(headerHashMap),
+                                            getRateResetFromHeader(headerHashMap), convertToPersonList(cloudPersons));
+    }
+
+    /**
+     * Returns the limit status information
+     *
+     * This does NOT consume API usage.
+     *
+     * @return
+     * @throws IOException if content cannot be interpreted
+     */
+    @Override
+    public ExtractedCloudResponse<RateLimitStatus> getLimitStatus() throws IOException {
+        RawCloudResponse cloudResponse = cloud.getRateLimitStatus();
+        HashMap<String, Long> headerHashMap = getHashMapFromHeader(cloudResponse.getHeaders());
+        if (!isValid(cloudResponse)) {
+            return getResponseWithNoData(cloudResponse, headerHashMap);
+        }
+        return new ExtractedCloudResponse<>(cloudResponse.getResponseCode(), getRateLimitFromHeader(headerHashMap),
+                                            getRateRemainingFromHeader(headerHashMap),
+                                            getRateResetFromHeader(headerHashMap), null);
+    }
+
+    @Override
+    public ExtractedCloudResponse<Void> createAddressBook(String addressBookName) throws IOException {
```

Comment:
This is an API not in GitHub. We require this since there is no way to add an addressbook without having to manually adding a cloud file. Should this also cost API usage?